src/lib/InstantSearch.ts

```diff
@@ -276,7 +276,7 @@ See ${createDocumentationLink({
     // middleware so they're notified of changes.
     if (this.started) {
       newMiddlewareList.forEach(m => {
-        m.subscribe();
+        m.subscribe?.();
```

Comment:
what about setting default noops in https://github.com/algolia/instantsearch.js/pull/4673/files#diff-3c46288a69cb2289a7369cff0f629366e99041bc5f60c586cdc29322130f9f84R272 instead? That way less code is conditional

Reply:
https://github.com/algolia/instantsearch.js/pull/4673/commits/02a01bec27e595c2dc8811a5acf75a34c78e9d02

I have a question.

1. `PartialMiddleware` and `Middleware`
vs
2. `Middleware` and `InternalMiddleware`

Is it okay to have the type of the public one having `Partial` or whatever prefix? Or should we rename it to `Middleware` and rename the internal one to something else?

Reply:
Middleware should probably be the public one (thus with optionals) I think